src/resource/resource-opening-hours/opening-period/PeriodOpeningHours.tsx

```diff
@@ -39,6 +44,40 @@ function renderStartAndEndTimes(
   }`;
 }
 
+function timeSpanDescriptionExistsInSomeLanguage(
+  timeSpanDescriptionObj: LanguageStrings | undefined
+): boolean {
+  if (!timeSpanDescriptionObj) {
+    return false;
+  }
+  return !!(
+    timeSpanDescriptionObj.fi ||
+    timeSpanDescriptionObj.sv ||
+    timeSpanDescriptionObj.en
+  );
+}
+
+function renderTimeSpanDescription(
+  timeSpanDescriptionObj: LanguageStrings | undefined,
+  language: Language
+): string {
+  if (!timeSpanDescriptionObj) {
+    return '';
+  }
+
+  if (timeSpanDescriptionExistsInSomeLanguage(timeSpanDescriptionObj)) {
+    if (!timeSpanDescriptionObj[language]) {
+      return displayLangVersionNotFound({
+        language,
+        label: 'aukiolon kuvaus',
+      });
+    }
+    return timeSpanDescriptionObj[language] || '';
+  }
+
+  return '';
+}
```

Comment:
Would it be simpler with:
```js
  if (!timeSpanDescriptionObj) {
      return '';
  }

  if (!timeSpanDescriptionObj[language]) {
     return displayLangVersionNotFound({
       language,
       label: 'aukiolon kuvaus',
     });
  }

  return timeSpanDescriptionObj[language]);
}
```

Reply:
Yes, but if the user has not set the description (optional) to any language on purpose, then would it be wrong to say that a specific version is missing? Hence the check for timeSpanDescriptionExistsInSomeLanguage hmm

Reply:
- [ ] Simplify function